src/cmap/auth/mongodb_oidc/automated_callback_workflow.ts

```diff
@@ -0,0 +1,100 @@
+import { type Document } from 'bson';
+import { setTimeout } from 'timers/promises';
+
+import { type Connection } from '../../connection';
+import { type MongoCredentials } from '../mongo_credentials';
+import {
+  OIDC_VERSION,
+  type OIDCCallbackFunction,
+  type OIDCCallbackParams,
+  type OIDCResponse
+} from '../mongodb_oidc';
+import { AUTOMATED_TIMEOUT_MS, CallbackWorkflow } from './callback_workflow';
+import { type TokenCache } from './token_cache';
+
+/** Must wait at least 100ms between invokations */
+const CALLBACK_DELAY = 100;
+
+/**
+ * Class implementing behaviour for the non human callback workflow.
+ * @internal
+ */
+export class AutomatedCallbackWorkflow extends CallbackWorkflow {
+  private lastInvokationTime: number;
```

Comment:
Similarly here, s/lastInvokationTime/lastInvocationTime/ (unless this is intentional? it feels like it might be 😅)

Reply:
I think I was thinking of this as something magical and german. :) I've changed.